--- a/build/appini_header.py
+++ b/build/appini_header.py
@@ -57,6 +57,11 @@
     else:
         appdata["App:sourceurl"] = "NULL"
 
+    if "App:UserAgentName" in appdata:
+        appdata["App:UserAgentName"] = '"%s"' % appdata["App:UserAgentName"]
+    else:
+        appdata["App:UserAgentName"] = "NULL"
+
     if "AppUpdate:url" not in appdata:
         appdata["AppUpdate:url"] = ""
 
@@ -75,7 +80,7 @@
                  "%(Gecko:maxversion)s",
                  "%(Crash Reporter:serverurl)s",
                  %(App:profile)s,
-                 NULL, // UAName
+                 %(App:UserAgentName)s,
                  %(App:sourceurl)s,
                  "%(AppUpdate:url)s"
              };"""
